pelican/plugins/show_source/show_source.py

```diff
@@ -1,59 +1,78 @@
 import logging
 import os
-from urllib.parse import urljoin
+import six
 
 from pelican import signals
 from pelican.utils import pelican_open
+from six.moves.urllib.parse import urljoin
+
+if not six.PY3:
+    from codecs import open
 
 logger = logging.getLogger(__name__)
 source_files = []
-PROCESS = ["articles", "pages", "drafts"]
-
+TYPES_TO_PROCESS = ['articles', 'pages']
 
 def link_source_files(generator):
     """
     Processes each article/page object and formulates copy from and copy
     to destinations, as well as adding a source file URL as an attribute.
     """
     # Get all attributes from the generator that are articles or pages
-    posts = [
+    documents = sum([
         getattr(generator, attr, None)
-        for attr in PROCESS
-        if getattr(generator, attr, None) is not None
-    ]
+        for attr in TYPES_TO_PROCESS
+        if getattr(generator, attr, None)
+    ], [])
+
+    autoext_setting = generator.settings.get(
+        'SHOW_SOURCE_AUTOEXT', False
```

Comment:
I assume this is a new setting that you're proposing to add? If so, shouldn't that be documented?